src/NativeScript/Marshalling/Reference/ExtVectorTypeInstance.cpp

```diff
@@ -79,15 +84,21 @@ const char* ExtVectorTypeInstance::encode(VM& vm, JSCell* cell) {
     return self->_compilerEncoding.c_str();
 }
 
-void ExtVectorTypeInstance::finishCreation(JSC::VM& vm, JSCell* innerType) {
+void ExtVectorTypeInstance::finishCreation(JSC::VM& vm, JSCell* innerType, bool isStructMember) {
     Base::finishCreation(vm);
     ffi_type* innerFFIType = const_cast<ffi_type*>(getFFITypeMethodTable(vm, innerType).ffiType);
 
     size_t arraySize = this->_size;
 
+#if defined(__x86_64__)
```

Comment:
This code makes the 3 element vector case be considered as 4 elements, in order to take into account this ABI peculiarity. In my opinion however, this is not the place for such a logic - it will be better suited in libffi where the ABI specification is considered and implemented.